contrib/win32/win32compat/shell-host.c

```diff
@@ -39,6 +39,19 @@
 #include "misc_internal.h"
 #include "inc\utf.h"
 
+/* Terminal Types */
+#define TERM_TYPE_ANSI    "ansi"
+#define TERM_TYPE_VT100   "vt100"
+#define TERM_TYPE_XTERM   "xterm"
+
+#define TT_ANSI     0x0001        /* Codes in all ANSI like terminals. */
+#define TT_VT100    0x0002        /* VT100 */
+#define TT_XTERM    0x0004        /* Xterm */
+
+#define TTM_VT100X   (TT_VT100 | TT_XTERM)
```

Comment:
> win32-openssh client is not working..

I did not specifically implement the support OpenSSH-Win32 client. This should make you.

> As per the description it fixes the scrolling and history.. from long time, history is working fine on all the terminals..

You are mistaken, because you have not correctly implemented the VT100 command **CUP**: https://github.com/PowerShell/Win32-OpenSSH/issues/724#issuecomment-302335471

> what changes are required for the "ansi" mode?

At the moment I'm at a loss to answer.